catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -415,17 +415,24 @@ public CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
     Stopwatch stopwatch = Stopwatch.createStarted();
 
     List<QueryResponse> responses = Collections.synchronizedList(new ArrayList<>());
-    QueryFunction queryFunction =
-        (queryRequest) -> {
-          QueryResponse queryResponse = catalogFramework.query(queryRequest);
-          responses.add(queryResponse);
-          return queryResponse;
-        };
-
-    List<Result> results =
-        ResultIterable.resultIterable(queryFunction, request, cqlRequest.getCount())
-            .stream()
-            .collect(Collectors.toList());
+
+    List<Result> results;
+    if (cqlRequest.getCount() == 0) {
```

Comment:
❓Just to make sure I understand, if `getCount() == 0`, we only want to get the number of "hits"? If that's accurate, should we change the `Query` object to only retrieve one result? We don't need to retrieve more than that to get the count.

Reply:
✏️ Might be worth extracting private methods with clear names (e.g., `retrieveHitCount()` and `retrieveResults()`) or add comments explaining what the `if` and `else` blocks do.

Reply:
That is accurate; but the framework does not respond with the actual results when it is a hit-count query, so this isn't necessary.  Based on the current implementation of CqlQuery, this might be a non-trivial change as well.

Reply:
separated the functionality into 2 separate private methods